lib/raygun/client.rb

```diff
@@ -140,7 +140,7 @@ def raw_data(rack_env)
 
         # If size is 0 the buffer is at best empty and at worst
         # something like the Puma::NullIO buffer which is missing methods
-        if input && input.size && !request.form_data?
+        if input && input.size && input.respond_to?(:pos) && !request.form_data?
```

Comment:
is there a way we could still receive the input data in the raygun tracking instead of skipping it?

Reply:
There's probably a way to make it work with the Unicorn buffer, but so far the `raw_data` method has been the most problematic one in the client so I really need to do another investigation and try to figure out a more reliable method of reading the raw request data. One that works across the various web servers.

Unfortunately, I'm a bit swamped with work currently so it will probably be a few weeks before I have time to look. If it's important I can talk to my manager about getting some work shuffled around to bump this up.

Did you just recently update to a newer version of Raygun4Ruby? I did some work a few months ago to make the raw_data method more reliable to fix an issue reported, but it's possible that broke it for your setup on Unicorn (or just unicorn in general).

Reply:
No recent upgrade. I think the only reason we noticed is that we happened to start getting a slew of webhook requests that were erroring, and raygun4ruby was choking on the buffer, causing our logs to blow up.

It's unfortunate we won't be able to get this working with our Unicorn setup as I believe we were one of the original raygun customers that requested this feature in the first place 😆

Reply:
Not the best solution :( I've had a talk with my manager but unfortunately, I probably won't be able to get any time to work on this until next week at the earliest, but probably not until the week after that. Would have time to look at a PR if one got sent my way before that, but otherwise, it will be a few weeks I'm afraid.

Was it reading the raw data before and it just started blowing up or has that been an issue for a while?

Reply:
My team is telling me it works/worked for some requests but does not for others. Sorry, I don't have more information right now.

Reply:
Definitely still looking forward to this!

Reply:
Hey @TSMMark, I've just pushed Raygun4Ruby v2.5.0 to RubyGems with better handling for raw_datamm. Should not break in this case and let you capture the raw_data again. Sorry it's taken so long!

Reply:
Thanks! Updating version now